Added Python 3 compatibility

BubaVV opened the pull request:
Now module support both Python 2.7 and 3.5+

BubaVV committed:
Add files via upload

Update for Python 3 compatibility

BubaVV committed:
Revert "Add files via upload"

This reverts commit 07b000a81001ab6af707f0450fccefffe6137fce.

BubaVV committed:
Did 2to3

Tests seems to work

BubaVV committed:
Now module compatible both with 2.7 and 3.x

BubaVV committed:
Build scripts now support 3.5

chadfurman commented:
I'm... gonna try this :p

chadfurman commented:
@BubaVV Thanks, it installed without issue (had to `pip3 install future` and `sudo python setup.py install` -- that's it).  I'm new to GAs so I won't be able to test it _too_ much.  But, if it works for my use-case that will be sweet.  I'll report back either way.

chadfurman commented:
Was able to make it through the first tutorial.  Copying their "final code" and adjusting the print statement for python3 (lol) and it worked fine

ranisalt commented:
@perone stop ignoring your repo

perone commented:
@ranisalt if you think that a pull-request with 51 files changed, without testing implemented, will be merged fast, then you'll need to review what you think that an open-source contribution is. However, you are always free to implement testing for the project and submit a pull-request. But for that you'll have to invest more effort than what you did for the complaint =)

lczyk commented:
@peronoe For testing I can recommend [Travis CI](https://github.com/marketplace/travis-ci).

chadfurman commented:
Perhaps @ranisalt will test it for us :)

Commit:
add future as setup dependency in setup.py

Commit:
remove future as setup dependency and add future as install dependency inside setup.py

BubaVV committed:
Merge pull request #1 from satizabal/master

Solving dependency with module future

BubaVV committed:
Py 3.6 and PEP8 envs added to Tox

Commit:
solving bug in GSimpleGA

Commit:
solving bug with exit

BubaVV committed:
Merge pull request #2 from satizabal/master

two small bugs...

Commit:
Add coverage support and tests for Statistics module

BubaVV committed:
PEP8 test added

Tox now supports PEP8 check via Flake. Examples are  fixed to pass test

BubaVV committed:
Everything else now passes PEP8

BubaVV committed:
Tests added for G1DBinaryString

Module 100% covered by tests. Deprecated __setslice__ method removed from G1DBase because it overrided by __setitem__

BubaVV committed:
Test for G2DList started

BubaVV committed:
Add tests for G2DLists

BubaVV committed:
Merge branch 'g2dlist_tests'

BubaVV committed:
All genomes are covered by tests

BubaVV committed:
Windows compatibility for tests

Also 3.6 tested a little bit

BubaVV committed:
Add tests for GAllele

BubaVV committed:
Travis CI added

Probably not working

BubaVV committed:
Travis CI fix

Coverage failed and turned off

BubaVV committed:
Convert Readme to Markdown
